app/src/main/kotlin/at/bitfire/davdroid/sync/AddressBookSyncer.kt

```diff
@@ -48,8 +52,13 @@ class AddressBookSyncer @AssistedInject constructor(
         get() = ContactsContract.AUTHORITY // Address books use the contacts authority for sync
 
 
-    override fun localSyncCollections(provider: ContentProviderClient): List<LocalAddressBook>
-        = LocalAddressBook.findAll(context, provider, account)
+    // FIXME should return _all_ address books; otherwise address book accounts of unchecked address books will not be removed
+    override fun localSyncCollections(provider: ContentProviderClient): List<LocalAddressBook> =
+        serviceRepository.getByAccountAndType(account.name, serviceType)?.let { service ->
+            getSyncCollections(service.id).mapNotNull { collection ->
+                LocalAddressBook.findByCollection(context, provider, collection.id)
+            }
+        }.orEmpty()
```

Comment:
Should return _all_ address books; otherwise address book accounts of unchecked address books will not be removed.

To try:

1. enable address book, sync
2. Corresponding address book account is created
3. disable address book, sync again
4. Corresponding address book account is still there (but should have been removed)

Reply:
Ah! I knew of the issue, but not why it happened. Nice. :)